app/src/main/java/atlant/moviesapp/presenters/UserFavoritesPresenter.java

```diff
@@ -0,0 +1,150 @@
+
+
+        call.enqueue(new Callback<MoviesResponse>() {
+            @Override
+            public void onResponse(Call<MoviesResponse> call, Response<MoviesResponse> response) {
+                int statusCode = response.code();
+                if (statusCode == 200) {
+                    List<Movie> movies = response.body().getResults();
+                    view.hideProgress();
+                    view.showMovies(movies);
+                } else {
+                    view.hideProgress();
+                    view.showError();
+                }
+
+
+            }
+
+            @Override
+            public void onFailure(Call<MoviesResponse> call, Throwable t) {
+                if (call.isCanceled()) {
+
+                    Log.e("TAG", "request was cancelled");
+                }
+                view.hideProgress();
+                view.showError();
+                //TODO onFailure
+                Log.e("TAG", t.toString());
+
+            }
+        });
+
+
+    }
+    public void getSeriesFavorites(int page) {
+
+        final ApiInterface apiservice = ApiClient.getClient().create(ApiInterface.class);
+        seriescall = apiservice.getUserFavoriteSeries(ApplicationState.getUser().getId(),API_KEY,ApplicationState.getUser().getSessionId(),"created_at.desc",page);
+
+
+        seriescall.enqueue(new Callback<TvShowsResponse>() {
+            @Override
+            public void onResponse(Call<TvShowsResponse> call, Response<TvShowsResponse> response) {
+                int statusCode = response.code();
+                if (statusCode == 200) {
+                    List<TvShow> shows = response.body().getResults();
+                    view.hideProgress();
+                    view.showTvShows(shows);
+                } else {
+                    view.hideProgress();
+                    view.showError();
+                }
+
+
+            }
+
+            @Override
+            public void onFailure(Call<TvShowsResponse> call, Throwable t) {
+                if (call.isCanceled()) {
+
+                    Log.e("TAG", "request was cancelled");
+                }
+                view.hideProgress();
+                view.showError();
+                //TODO onFailure
+                Log.e("TAG", t.toString());
+
+            }
+        });
+
+
+    }
+    public void postFavorite(int id, String session_id, BodyFavourite favorite) {
+        final ApiInterface apiservice = ApiClient.getClient().create(ApiInterface.class);
+        postCall = apiservice.addFavorite(id, API_KEY, session_id, favorite);
+        postCall.enqueue(new Callback<PostResponse>() {
+            @Override
+            public void onResponse(Call<PostResponse> call, Response<PostResponse> response) {
+                int statusCode = response.code();
+                if (statusCode == 201) {
+
+
+                } else {
+                    //TODO onFailure
+                    Log.e("TAG", "Error");
```

Comment:
Later, no need in this PR, before pushing the code to github we should remove all Logs